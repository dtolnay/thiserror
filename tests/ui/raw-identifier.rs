use thiserror::Error;

#[derive(Error, Debug)]
#[error("error: {r#fn}")]
pub struct Error {
    r#fn: &'static str,
}

fn main() {
    let r#fn = "...";
    let _ = format!("error: {r#fn}");
}
