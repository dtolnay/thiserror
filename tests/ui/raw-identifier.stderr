error: invalid format string: raw identifiers are not supported
 --> tests/ui/raw-identifier.rs:4:18
  |
4 | #[error("error: {r#fn}")]
  |                  --^^
  |                  |
  |                  raw identifier used here in format string
  |                  help: remove the `r#`
  |
  = note: identifiers in format strings can be keywords and don't need to be prefixed with `r#`

error: invalid format string: raw identifiers are not supported
  --> tests/ui/raw-identifier.rs:11:30
   |
11 |     let _ = format!("error: {r#fn}");
   |                              --^^
   |                              |
   |                              raw identifier used here in format string
   |                              help: remove the `r#`
   |
   = note: identifiers in format strings can be keywords and don't need to be prefixed with `r#`
